cockpit/IPC/src/Components/Table/Table.tsx

```diff
@@ -0,0 +1,885 @@
+         onChange={(_event, value) => onSearchChange(value)}
+         onClear={() => onSearchChange('')}
+      />
+   );
+
+   // Set up Type single select
+   const [isTypeMenuOpen, setIsTypeMenuOpen] = React.useState<boolean>(false);
+   const typeToggleRef = React.useRef<HTMLButtonElement>(null);
+   const typeMenuRef = React.useRef<HTMLDivElement>(null);
+   const typeContainerRef = React.useRef<HTMLDivElement>(null);
+
+   const handleTypeMenuKeys = (event: KeyboardEvent) => {
+      if (isTypeMenuOpen && typeMenuRef.current?.contains(event.target as Node)) {
+         if (event.key === 'Escape' || event.key === 'Tab') {
+            setIsTypeMenuOpen(!isTypeMenuOpen);
+            typeToggleRef.current?.focus();
+         }
+      }
+   };
+
+   const handleTypeClickOutside = (event: MouseEvent) => {
+      if (isTypeMenuOpen && !typeMenuRef.current?.contains(event.target as Node)) {
+         setIsTypeMenuOpen(false);
+      }
+   };
+
+   React.useEffect(() => {
+      window.addEventListener('keydown', handleTypeMenuKeys);
+      window.addEventListener('click', handleTypeClickOutside);
+      return () => {
+         window.removeEventListener('keydown', handleTypeMenuKeys);
+         window.removeEventListener('click', handleTypeClickOutside);
+      };
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, [isTypeMenuOpen, typeMenuRef]);
+
+   const onTypeToggleClick = (ev: React.MouseEvent) => {
+      ev.stopPropagation();
+      setTimeout(() => {
+         if (typeMenuRef.current) {
+            const firstElement = typeMenuRef.current.querySelector('li > button:not(:disabled)');
+            firstElement && (firstElement as HTMLElement).focus();
+         }
+      }, 0);
+      setIsTypeMenuOpen(!isTypeMenuOpen);
+   };
+
+   function onTypeSelect(event: React.MouseEvent | undefined, itemId: string | number | undefined) {
+      if (typeof itemId === 'undefined') {
+         return;
+      }
+
+      setTypeSelection(itemId.toString());
+      setIsTypeMenuOpen(!isTypeMenuOpen);
+   }
+
+   const typeToggle = (
+      <MenuToggle
+         ref={typeToggleRef}
+         onClick={onTypeToggleClick}
+         isExpanded={isTypeMenuOpen}
+         style={
+            {
+               width: '200px'
+            } as React.CSSProperties
+         }
+      >
+         Filter by Type
+      </MenuToggle>
+   );
+
+   const typeMenu = (
+      <Menu ref={typeMenuRef} id="attribute-search-type-menu" onSelect={onTypeSelect} selected={typeSelection}>
+         <MenuContent>
+            <MenuList>
+               <MenuItem itemId="unknown">unknown</MenuItem>
+               <MenuItem itemId="bool">bool</MenuItem>
+               <MenuItem itemId="int64_t">int64_t</MenuItem>
+               <MenuItem itemId="uint64_t">uint64_t</MenuItem>
+               <MenuItem itemId="double">double</MenuItem>
+               <MenuItem itemId="string">string</MenuItem>
+               <MenuItem itemId="json">json</MenuItem>
```

Comment:
Can we auto populate this from the signals that we get?

more will be added in the future and this also adds cluter to small systems with few types.